lib/db/schema/patch-018-019.sql

```diff
@@ -0,0 +1,199 @@
+
+CREATE TABLE devices (
+  uid BINARY(16) NOT NULL,
+  id INT UNSIGNED NOT NULL,
+  sessionTokenId BINARY(32),
+  name VARCHAR(255) NOT NULL,
+  type VARCHAR(16) NOT NULL,
+  createdAt BIGINT UNSIGNED NOT NULL,
+  callbackURL VARCHAR(255),
+  INDEX devices_uid (uid),
+  PRIMARY KEY (uid, id)
+) ENGINE=InnoDB;
+
+
+CREATE PROCEDURE `createDevice_1` (
+    IN `inUid` BINARY(16),
+    IN `inSessionTokenId` BINARY(32),
+    IN `inName` VARCHAR(255),
+    IN `inType` VARCHAR(16),
+    IN `inCreatedAt` BIGINT UNSIGNED,
+    IN `inCallbackURL` VARCHAR(255)
+)
+BEGIN
+  INSERT INTO devices(
+    uid,
+    id,
+    sessionTokenId,
+    name,
+    type,
+    createdAt,
+    callbackURL
+  )
+  SELECT
+    inUid,
+    COALESCE(MAX(id), 0) + 1,
```

Comment:
To clarify, the idea here is to to ensure that each user gets devices numbered `1`, `2`, `3` etc?  I seem to recall MySQL having a feature that can do this with auto-increment columns if they're part of a compound primary key, which may be both more efficient and clearer.  (That is to say, if you just declare the `id` column to be auto-increment, MySQL may magically do what you want).